src/mesh.js

```diff
@@ -300,6 +300,169 @@ export default class Mesh {
         self.materialsByIndex = {};
     }
 
+    calculateTangentsAndBitangents() {
+        console.assert(this.vertices && this.vertices.length &&
+            this.vertexNormals && this.vertexNormals.length &&
+            this.textures && this.textures.length, "Missing attributes for calculating tangents and bitangents");
+
+        const unpacked = {};
+        unpacked.tangents = [...new Array(this.vertices.length)].map(v => 0); 
+        unpacked.bitangents = [...new Array(this.vertices.length)].map(v => 0); 
+
+        // Loop through all faces in the whole mesh
+        let indices;
+        // If sorted by material
+        if (Array.isArray(this.indices[0])) {
+            debugger;
```

Comment:
Don't forget to remove this `debugger;` statement.

Reply:
Oops!

Reply:
Oops!